libs/components/src/lib/search-bar/search-bar.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useState } from 'react';
+import ReactSearchBox from 'react-search-box';
+import './search-bar.css';
+
+interface SearchBarProps<T> {
+  onSearch: (val: string) => T;
+}
+
+export function SearchBar<T>(props: SearchBarProps<T>) {
+  const [searchVal, setSearchVal] = useState('');
+  const recommendedResultsData = [
+    {
+      key: 'ginger',
+      value: 'ginger for nausea',
+    },
+  ];
+
+  return (
+    <div className="search-bar-container">
+      <ReactSearchBox
+        placeholder="Your Search"
+        aria-label="search bar"
+        value={searchVal}
+        data={recommendedResultsData}
+        onChange={(val) => setSearchVal(val)}
+        callback={(record) => console.log(record)}
+      />
+      <button aria-label="search button" onClick={() => props.onSearch(searchVal)}>Search</button>
```

Comment:
Aria-labels are pretty tricky, I'll share a deck with you all on some basics here.
But basically, this button shouldn't need an aria-label, basically because you are using the <button> native element, and gave the button the text of "Search", you've already set it up for accessibility purposes and don't require an extra aria-label! This combination of <button> and setting the text to "Search" allows screen readers to announce to users that the element they have selected is of type "button", and will read the button label (which you set to "Search")

Reply:
Ohh, ok cool!